### ISSUE TITLE
Route not recognized

### ISSUE THREAD

**Issue by knownasilya** [denali-js/denali#154]
I've been seeing the following issue with new apps:

```
[2016-10-12T14:42:16.144Z] ERROR - NotFoundError: Route not recognized
    at Router.handle (/Users/iradchenko/sandbox/denali/dist/lib/runtime/router.js:143:52)
    at emitTwo (events.js:106:13)
    at Server.emit (events.js:191:7)
    at HTTPParser.parserOnIncoming [as onIncoming] (_http_server.js:546:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23)
```

To reproduce create a new app, `npm start`, visit `localhost:3000` and it should happen on the first time, although the route loads fine.. maybe it's an issue with the appending of the '/'?

**Comment by acorncom** [comment]
Odd, haven't seen this one myself (though I'm working with an existing app at this point ...)

**Comment by davewasmer** [comment]
@knownasilya are you still able to reproduce?

**Comment by knownasilya** [comment]
Still an issue with a brand new app, `denali new hello && cd hello && denali server` then open `localhost:3000`.

**Comment by knownasilya** [comment]
```
Installing dependencies ... done ✔
Setting up git repo ... done ✔
🏔 Installation complete

To launch your application, just run:

  $ cd hello && denali server


 ▲ ~/battleground cd hello 

 ▲ ~/battleground/hello denali server                                                                                                              ⇡ master :: 0m :: ⬢
Build successful
[2016-10-23T18:14:42.509Z] INFO - hello@0.0.1 server up on port 3000
get / 200 15.293 ms - -
[2016-10-23T18:14:51.969Z] ERROR - NotFoundError: Route not recognized
    at Router.handle (/Users/iradchenko/battleground/hello/node_modules/denali/lib/runtime/router.js:92:50)
    at emitTwo (events.js:106:13)
    at Server.emit (events.js:191:7)
    at HTTPParser.parserOnIncoming [as onIncoming] (_http_server.js:546:12)
    at HTTPParser.parserOnHeadersComplete (_http_common.js:99:23
```

**Comment by davewasmer** [comment]
Hm, I'm unable to reproduce this now. Perhaps the latest changes fixed it? Could you try again?

![image](https://cloud.githubusercontent.com/assets/707213/19745175/ce0c2b5c-9b8d-11e6-84d0-262bd7fb3b68.png)

**Comment by knownasilya** [comment]
Looks fixed to me.

**Comment by knownasilya** [comment]
Still seeing this.

**Comment by davewasmer** [comment]
What versions (i.e. what is your global version, and what version was installed in the newly generated app)?

**Comment by knownasilya** [comment]
Both are latest commit from master

**Comment by acburdine** [comment]
@davewasmer @knownasilya I just created a new app and am getting this error. What's interesting is it looks like there's a hardcoded path in there:

```sh
[2017-02-03T17:47:59.416Z] ERROR - NotFoundError: Route not recognized
    at Router.handle$ (/Users/daw/projects/oss/denali/denali/lib/runtime/router.js:101:15)
    at tryCatch (/Users/acburdine/Sites/tourdefours/api/node_modules/regenerator-runtime/runtime.js:64:40)
    at GeneratorFunctionPrototype.invoke [as _invoke] (/Users/acburdine/Sites/tourdefours/api/node_modules/regenerator-runtime/runtime.js:355:22)
    at GeneratorFunctionPrototype.prototype.(anonymous function) [as next] (/Users/acburdine/Sites/tourdefours/api/node_modules/regenerator-runtime/runtime.js:116:21)
    at tryCatch (/Users/acburdine/Sites/tourdefours/api/node_modules/regenerator-runtime/runtime.js:64:40)
    at invoke (/Users/acburdine/Sites/tourdefours/api/node_modules/regenerator-runtime/runtime.js:154:20)
    at /Users/acburdine/Sites/tourdefours/api/node_modules/regenerator-runtime/runtime.js:164:13
```

This line looks suspicious:
> at Router.handle$ (/Users/daw/projects/oss/denali/denali/lib/runtime/router.js:101:15)

Not sure why that filepath would show up but will do some investigating.

**Comment by davewasmer** [comment]
@acburdine that path is likely from sourcemaps.

Anytime you see a stack trace that points to a denali file (either denali itself, or your app's own files, or an addon's files) that does _not_ contain `dist`, it means thats the sourcemapped stack.

I'll have to update the build process to remove those, but they shouldn't be causing any harm I imagine - the line numbers should be valid and everything.

Did you see this error when you started the app, or when you made a request?

**Comment by acburdine** [comment]
I only noticed it once the very first time I made a request to a new Denali app, restarting the serve and requesting it again doesn't do it.

Steps:
1. `denali new <name>`
2. `denali serve`
3. `curl localhost:3000`
4. Observe: request is returned correctly "{message: "Welcome to Denali}", but the error is output to console.
5. Stop serve
6. Run `denali serve` again
7. `curl localhost:3000`
8. Observe: message is correctly output, no error output

On a whim I removed the `dist` and `tmp` directories too to see if it had anything to do with the build but I was only able to get the error once on a new app.
